sympy/sets/dispatchers.py

```diff
@@ -150,6 +241,27 @@ def function_sets(f, x):
         return Interval(log(x.start), log(x.end), x.left_open, x.right_open)
     return ImageSet(Lambda(_x, f(_x)), x)
 
-@dispatch(FunctionClass, Set)
+@dispatch(FunctionUnion, Union)
+def function_sets(f, x):
+    return Union(imageset(f, arg) for arg in x.args)
+
+@dispatch(FunctionUnion, Intersection)
+def function_sets(f, x):
+    # If the function is invertible, intersect the maps of the sets.
+    u = symbols("u")
+    fdiff = f(u).diff(u)
+    # TODO: find a better condition for invertible functions:
+    if ((f in (exp, log))  # functions known to be invertible
+        or (fdiff > 0) == True or (fdiff < 0) == True  # monotonous funcs
+        ):
```

Comment:
@jksuom I put these conditions of invertibility. `solveset` needs this mapping to solve equations with `Abs` and similar. It does not look that clean.

Do we have a better way to test the invertibility of functions?

Reply:
We probably should have a method for testing invertibility in a set such as the union of `x.args`.